cpp/src/Ice/ConnectRequestHandler.cpp

```diff
@@ -13,35 +13,25 @@
 #include <Ice/Protocol.h>
 #include <Ice/Properties.h>
 #include <Ice/ThreadPool.h>
+#include <Ice/ProxyFactory.h>
 
 using namespace std;
 using namespace IceInternal;
 
-ConnectRequestHandler::ConnectRequestHandler(const ReferencePtr& ref, const Ice::ObjectPrxPtr& proxy) :
+ConnectRequestHandler::ConnectRequestHandler(const ReferencePtr& ref) :
     RequestHandler(ref),
-    _proxy(proxy),
     _initialized(false),
     _flushing(false)
 {
 }
 
 RequestHandlerPtr
-ConnectRequestHandler::connect(const Ice::ObjectPrxPtr& proxy)
+ConnectRequestHandler::connect()
 {
     unique_lock lock(_mutex);
-    if(!initialized(lock))
-    {
-        _proxies.insert(proxy);
-    }
     return _requestHandler ? _requestHandler : shared_from_this();
```

Comment:
Seems that we never set `_requestHandler`

Reply:
You're right: I remove this _requestHandler and even the enclosing connect function.